src/NamespaceManager.php

```
@@ -15,275 +20,99 @@
+		$instance = new self();
+		$vars = $GLOBALS;
+		$localLanguage = $instance->localLanguage->fetch( $instance->getLanguageCode( $vars ) );
+		$extraNamespaces = $localLanguage->getNamespaces();
+		$namespaceAliases = $localLanguage->getNamespaceAliases();
+		$canonicalNames = self::getCanonicalNames();
 
-		// https://phabricator.wikimedia.org/T160665
-		// Find any namespace that uses the same canonical name and remove it
+		// T160665: drop any existing namespace that uses a name we own.
+		$namespacesByName = array_flip( $namespaces );
 		foreach ( $canonicalNames as $id => $name ) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$id'. (undefined)

(UnusedLocalVariable)